form/src/main/kotlin/com/ritense/form/service/PrefillFormService.kt

```diff
@@ -86,7 +90,16 @@ class PrefillFormService(
         val formDefinition = formDefinitionService.getFormDefinitionById(formDefinitionId)
             .orElseThrow { RuntimeException("Form definition not found by id $formDefinitionId") }
         if (documentId != null) {
-            val document = runWithoutAuthorization { documentService.get(documentId.toString()) }
+            val document = runWithoutAuthorization {
```

Comment:
I think you could just remove the `runWithoutAuthorization` here. Then you won't need the `authorizationService`